docs/adr/0016-use-graphql-and-apollo-for-settings-redesign.md

```diff
@@ -0,0 +1,135 @@
+
+### Option B - Layer GraphQL and Apollo on top of our REST architecture, making direct database calls for basic operations
+
+#### Description
+
+- This option allows us to begin working with GraphQL and Apollo on top of our REST architecture, proxying existing auth-server calls, except where it makes sense to interact with the database directly for simple operations.
+
+##### Pros:
+
+- Less risk and less upfront work than option C.
+- We'll still heavily reduce the number of initial network requests performed on the Settings frontend with an incremental approach. The Apollo Server can orchestrate the REST requests sufficient to fulfil just that query and return exactly the data requested.
+- GQL can proxy to existing auth-server calls. We can incrementally swap out auth-server API calls with direct implementation.
+
+##### Cons:
+
+- Creates tech debt. We won't be fully integrated until we finish swapping out the auth-server API calls.
+- Our stack will be less universal. We'll use REST for non-Settings FxA pages and for the payments server until refactoring can be done.
+- While GraphQL is database agnostic, we will be unable to use [GraphQL Subscriptions](https://www.apollographql.com/docs/react/data/subscriptions/) because our database does not support real-time data.
+
+### Option C - Full GraphQL and Apollo integration with direct database calls
+
+#### Description
+
+- This option would extend option B. The more complex operations routed through the auth-server would be directly implemented through the Apollo server.
+
+##### Pros
+
+- Less tech debt later.
+- We would see a performance boost over option B.
+- An all-out dedication to this approach could open the door to evaluating our database options (and being able to work with [GraphQL subscriptions](https://www.apollographql.com/docs/react/data/subscriptions/)) or more quickly pave a path to move our backend off of MySQL to something else for a more regional conscious architecture.
+
+##### Cons
+
+- Increased risk. While we'll be pushing the new Settings app to staging and production while we're building it out, the Settings Redesign project already has a lot of moving parts and an incremental approach definitely feels more ideal.
+- Much more time needed for setup, and access control, auth, and security already implemented by auth-server might be expensive to reinvent.
+
+## Additional Links
+
+- [Settings Redesign project](https://jira.mozilla.com/browse/FXA-840) (Jira)
+- [ADR for Settings Redesign as a new React App](https://github.com/mozilla/fxa/blob/master/docs/adr/0011-create-new-react-app-for-settings-redesign.md)
+- [GraphQL docs](https://graphql.org/learn/)
+- [Apollo docs](https://www.apollographql.com/docs/intro/platform/)
+- [GraphQL subscriptions](https://www.apollographql.com/docs/react/data/subscriptions/)
```

Comment:
Do we want to add a link to Ben's "GraphQL API Technical Specification"?

Reply:
Ah yes, good call, I should add that doc to the Redesign epic too.